src/commands/fun/poll.ts

```diff
@@ -1,111 +1,197 @@
 import { BOT } from '@root/config';
-import { Message, MessageEmbed, MessageReaction } from 'discord.js';
+import { ApplicationCommandOptionData, ButtonInteraction, CommandInteraction, Message, MessageActionRow, MessageButton, MessageEmbed } from 'discord.js';
 import parse from 'parse-duration';
 import { Command } from '@lib/types/Command';
 
-const QUESTION_CHAR_LIMIT = 1024;
-const EMBED_TITLE_CHAR_LIMIT = 256;
+const QUESTION_CHAR_LIMIT = 256;
+const args = ['Single', 'Multiple'];
 
 export default class extends Command {
 
-	description = `Have ${BOT.NAME} create a poll for you`;
-	aliases = ['vote'];
-	usage = '<timespan>|<question>|<choices...>';
-	extendedHelp = 'You can have two to ten choices.';
+	description = `Have ${BOT.NAME} create a poll for you.`;
+
+	options: ApplicationCommandOptionData[] = [
+		{
+			name: 'timespan',
+			description: `How long your poll should last. Acceptable formats include '5s', '5m', '5h', '5h30m', '7h30m15s'...`,
+			type: 'STRING',
+			required: true
+		},
+		{
+			name: 'question',
+			description: `What would you like to ask?`,
+			type: 'STRING',
+			required: true
+		},
+		{
+			name: 'choices',
+			description: `A poll can have 2-10 choices. Separate choices with '|' (no spaces/quotes).`,
+			type: 'STRING',
+			required: true
+		},
+		{
+			name: 'optiontype',
+			description: `Whether participants can only select one choice or multiple.`,
+			type: 'STRING',
+			required: true,
+			choices: args.map((arg) => ({
+				name: arg,
+				value: arg
+			}))
+		}
+	]
 	runInDM = false;
 
+	run(_msg: Message): Promise<void> { return; }
+
+	resetArray(array: number[], len: number): number[] {
```

Comment:
`Array.fill();` ([docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/fill)) accomplishes this in one line